src/ansiblelint/generate_docs.py

```diff
@@ -3,7 +3,7 @@
 from typing import Iterable
 
 from rich import box
-from rich.console import render_group
+from rich.console import group
```

Comment:
Maybe we should do something like this:

```suggestion
# Remove this compatibility try-catch block once we drop support for rich < 10.7.0
try:
    from rich.console import group
except ImportError:
    from rich.console import render_group as group
```

Reply:
I don't like do the try/except dance. But, if that's what we want I'll change.

Reply:
I am also not a fan of this, but there is not much we can do here since upstream is not taking backward compatibility seriously enough.

Reply:
At least they bumped the major version. Good part is that this did not affect normal usage of the linter because that module is used only to generate docs.

Reply:
It definitely broke our usage of the linter

https://github.com/spantaleev/matrix-docker-ansible-deploy/runs/4757647392?check_suite_focus=true
https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/7d96526b531a6ffdda79c081663a81e8ffc234ab/.github/workflows/ansible-lint.yml

And I'm guessing all of the other people who 👍ed https://github.com/ansible-community/ansible-lint/issues/1795 weren't all trying to generate docs.

Reply:
Actually, the rich version bump did screw us over in all cases because we load that module at https://github.com/ansible-community/ansible-lint/blob/main/src/ansiblelint/__main__.py#L214 and not inside the relevant `if` clause at https://github.com/ansible-community/ansible-lint/blob/main/src/ansiblelint/__main__.py#L219